tests/unit/backend/corpora/api_server/curator/collection/test_assets.py

```diff
@@ -0,0 +1,139 @@
+from unittest.mock import patch, Mock
+
+from backend.corpora.common.corpora_orm import CollectionVisibility, DatasetArtifactFileType
+from tests.unit.backend.corpora.api_server.base_api_test import BaseAuthAPITest
+from tests.unit.backend.fixtures.config import fake_s3_file
+from tests.unit.backend.fixtures.mock_aws_test_case import CorporaTestCaseUsingMockAWS
+
+
+# TODO test with dataset_uuid = test_curator_tag
+
+
+class TestAsset(BaseAuthAPITest, CorporaTestCaseUsingMockAWS):
+    def setUp(self):
+        # Needed for proper setUp resolution in multiple inheritance
+        super().setUp()
+        self.test_dataset_uuid = ["test_dataset_id", "test_curator_tag"]
+        self.curator_tag = "curator_tag"
+        self.generate_collection(
+            self.session,
+            id="test_curator_tag_collection_id",
+            visibility=CollectionVisibility.PUBLIC.name,
+            owner="test_user_id",
+            name="test_collection_name",
+            description="test_description",
+            data_submission_policy_version="0",
+            contact_name="Some Body",
+            contact_email="somebody@chanzuckerberg.com",
+        )
+        self.generate_dataset(
+            self.session,
+            id="test_curator_tag",
+            curator_tag="curator_tag",
+            revision=0,
+            name="test_dataset_name",
+            schema_version="2.0.0",
+            collection_id="test_curator_tag_collection_id",
+            artifacts=[
+                dict(
+                    filename="test_filename",
+                    filetype=DatasetArtifactFileType.H5AD.name,
+                    user_submitted=True,
+                    s3_uri=fake_s3_file,
+                )
+            ],
+        )
+
+    def test__get_dataset_asset__OK(self):
+        bucket = self.CORPORA_TEST_CONFIG["bucket_name"]
+        s3_file_name = "test_s3_uri.h5ad"
+        content = "Hello world!"
+        self.create_s3_object(s3_file_name, bucket, content=content)
+
+        for dataset_uuid in self.test_dataset_uuid:
+            with self.subTest(dataset_uuid):
+                expected_body = dict(
+                    dataset_uuid=dataset_uuid,
+                    assets=[dict(file_name="test_filename", file_size=len(content), file_type="H5AD")],
+                )
+                if dataset_uuid == "test_curator_tag":
+                    expected_body["curator_tag"] = self.curator_tag
+                    response = self.app.get(
+                        "/curation/v1/collections/test_curator_tag_collection_id/assets",
+                        query_string=dict(curator_tag=self.curator_tag),
+                    )
+                else:
```

Comment:
should add `expected_body["curator_tag"] = None`?

Reply:
curator_tag is not a required field in the response, so it's not returned if it doesn't exist.

Reply:
Got it - I think this gets back to the issue Andrew and Brian were discussing recently about the shapes of our responses (including null values or no). Speaking of which, I'm not sure we ever came to a clear resolution...